rmpv/src/lib.rs

```diff
@@ -1006,6 +1007,89 @@ where V: Into<Value> {
   }
 }
 
+impl TryFrom<Value> for u64 {
+  type Error = Value;
+
+  fn try_from(val: Value) -> Result<Self, Self::Error> {
+      match val {
+        Value::Integer(n) => {
+          match n.as_u64() {
+            Some(i) => Ok(i),
+            None => Err(val)
+          }
+        }
+        v => Err(v),
+      }
+  }
+}
+
+impl TryFrom<Value> for i64 {
+  type Error = Value;
+
+  fn try_from(val: Value) -> Result<Self, Self::Error> {
+      match val {
+        Value::Integer(n) => {
+          match n.as_i64() {
+            Some(i) => Ok(i),
+            None => Err(val)
+          }
+        }
+        v => Err(v),
+      }
+  }
+}
+
+impl TryFrom<Value> for f64 {
+  type Error = Value;
+
+  fn try_from(val: Value) -> Result<Self, Self::Error> {
+      match val {
+        Value::Integer(n) => {
+          match n.as_f64() {
+            Some(i) => Ok(i),
+            None => Err(val)
+          }
+        }
+        Value::F32(n) => Ok(From::from(n)),
+        Value::F64(n) => Ok(n),
+        v => Err(v),
+      }
+  }
+}
+
+impl TryFrom<Value> for (i8, Vec<u8>) {
```

Comment:
That is not obvious. I'd rather not add support for it this way. If `Ext` was a struct, it could be conversion to that struct directly. Otherwise it's best to leave it out.

Reply:
It's of no personal interest to me, but I'd like to understand your objection (it would feel like an omission somewhat). What's not obvious here? If you have a single value, and want an `(i8, Vec<u8>)` from it, it must be the case that the value is of variant `Ext`, which is part of the spec. Or am I misunderstanding something here?

Reply:
* It changes a strictly-typed thing to an anonymous type. To me there's nothing that says `Ext` is a tuple, and that it's a tuple with this particular order and types of fields. I wouldn't be able to guess what type I should use for `Ext`.
* It could as well mean conversion from `[Value; 2]` array to a tuple. Rust may get variadic tuples in the future.

Reply:
Ok, I see where you're coming from. I'd argue that the definition of `Ext` is part of the spec, so there's no guesswork involved, and also that it's `TryFrom<Value>`, not `TryFrom<[Value; 2]>`, so there's no confusion as well.

But no need to discuss, I'll remove this impl tonight, and put in a link or a comment explaining it.